### ISSUE TITLE
GET /api/v1/auth_providers/{id}/affiliates - may return empty email addresses

### ISSUE THREAD

**Issue by dleehr** [Duke-Translational-Bioinformatics/duke-data-service#1323]
When searching for users (affiliates) via the `/auth_providers/:provide_id/affiliates/` endpoint (called via [DukeDSClient's ddsapi.py](https://github.com/Duke-GCB/DukeDSClient/blob/48c9f8ba1f87a50bf97cb2756b2d2c6939108800/ddsc/core/ddsapi.py#L975), returned affiliate records have blank email addresses if the user has their "Set who can see your email address"  set to "Restricted" at https://idms-web.oit.duke.edu/portal/web:

![Screen Shot 2019-05-06 at 1 18 07 PM](https://user-images.githubusercontent.com/2752599/57242452-875fee00-7001-11e9-8c98-8bd0bfba263b.png)

This issue came up at the March 3 meeting demonstrating [DataDelivery](https://datadelivery.genome.duke.edu), but I didn't create an issue at the time. I believe the explanation was that his API is backed by an un-authenticated LDAP search, which will be treated as an external directory search.

**Comment by dmlond** [comment]
That is correct. Any request for an affiliate that does not already have a DDS account will defer to a search of the Duke LDAP system, which is governed by rules which allow a user to hide their email. The user can update their email in the DDS system after they login for the first time.

**Comment by dleehr** [comment]
> ...the Duke LDAP system, which is governed by rules which allow a user to hide their email. 

I agree we should respect this preference and not provide email addresses outside of Duke, but all parties involved are authenticated as Duke personnel.

> The user can update their email in the DDS system after they login for the first time.

We have a service account that I'd love to update this for - can you point me to it?

**Comment by dmlond** [comment]
Actually, we do not currently support user email updates :(
